### PR TITLE
[mlir][memref] Add memref alias folders for expand/collapse_shape for vector load/store

### PR DESCRIPTION
**Pull request by Groverkss**
This patch adds adds patterns to fold memref alias for expand_shape/collapse_shape feeding into vector.load/vector.store and vector.maskedload/vector.maskedstore

### CODE REVIEW

**Review by zero9178 — APPROVED** [review]
Generally LGTM with comments addressed.

Is the ´[arith]` in the title on purpose or just a copy-paste error or typo? Given that these patterns are in MemRef. 

Side node: Its funny to see type switches when its already templated on the operation type and could be `if constexpr`s. Don't view this as an actionable item tho :)

**Comment by Groverkss** [comment]
> Is the ´[arith]` in the title on purpose or just a copy-paste error or typo? Given that these patterns are in MemRef.

Typo, sorry. I don't know where the arith came from haha.

File: mlir/lib/Dialect/MemRef/Transforms/FoldMemRefAliasOps.cpp [hunk 1]
```diff
@@ -518,10 +518,25 @@ LogicalResult LoadOpOfExpandShapeOpFolder<OpTy>::matchAndRewrite(
...
-        rewriter.replaceOpWithNewOp<decltype(op)>(
+      .Case([&](affine::AffineLoadOp op) {
+        rewriter.replaceOpWithNewOp<affine::AffineLoadOp>(
             loadOp, expandShapeOp.getViewSource(), sourceIndices);
       })
+      .Case([&](memref::LoadOp op) {
+        rewriter.replaceOpWithNewOp<memref::LoadOp>(
+            loadOp, expandShapeOp.getViewSource(), sourceIndices,
+            op.getNontemporal());
+      })
```
[the remark this answers is not in the document]
Comment (Groverkss) [reply mlir/lib/Dialect/MemRef/Transforms/FoldMemRefAliasOps.cpp:529]: updated all tests touching these patterns + memref.load/store vector.load/store to have a nontemporal attribute and check if they are preserved.

**Review by zero9178 — APPROVED** [review]

**Comment by Groverkss** [comment]
Looks like the linux CI is stuck (the previous one was stuck for 3h). This issue should be platform independent and the windows CI passes, so I'm landing this.

**Pull request merged**